src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,50 +1,163 @@
 package core.basesyntax;
 
+import java.util.Iterator;
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
+    private static class Node<T> {
+        T value;
+        Node<T> next;
+        Node<T> prev;
+
+        public Node(T value, Node<T> next, Node<T> prev) {
+            this.value = value;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        addNodeToTheEnd(value);
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("This index is out of bounds.");
+        }
+        if (index == size) {
+            addNodeToTheEnd(value);
+        } else {
+            Node<T> soughtForNode = findNodeByIndex(index);
+            Node<T> previousNode = soughtForNode.prev;
+            Node<T> newNode = new Node<>(value, soughtForNode, previousNode);
+            soughtForNode.prev = newNode;
+            if (previousNode == null) {
+                firstNode = newNode;
+            } else {
+                previousNode.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        if (list.size() == 0) {
+            return false;
+        }
+        Iterator iterator = list.iterator();
+        while (iterator.hasNext()) {
+            add((T) iterator.next());
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> soughtForNode = findNodeByIndex(index);
+        return soughtForNode.value;
     }
 
+    // DOES THIS METHOD REPLACE THE NODE OR SHIFT IT?
```

Comment:
this method should replace the value of the node)
let's remove this comment from code